src/internal/operations/backup.go

```diff
@@ -709,6 +785,20 @@ func mergeDetails(
 			item := entry.ItemInfo
 			details.UpdateItem(&item, newLoc)
 
+			// If this item was modified in a later assist backup, update it
+			// with the assist backup's info.
+			assistDeets, ok := repoRefToAssistDeets[entry.RepoRef]
+			if ok && assistDeets.ItemInfo.Modified().After(item.Modified()) {
+				logger.Ctx(ctx).Infow("sourcing item info from assist backup",
+					"repo_ref", entry.RepoRef)
+
+				// If something is considered cached then we assume the mod time
+				// hasn’t changed. If the mod time hasn’t changed then we
+				// assume neither the item data nor metadata changed, hence it’s
+				// safe to source the item’s detail entry from assist deets.
+				item = assistDeets.ItemInfo
+			}
+
```

Comment:
I don't think this addition is needed. With how assist base details merging is implemented, the items it's merging with are already in deets but not in `dataFromBackup`. This function and the helper function(s) it calls should only interact with `dataFromBackup` to see if something should be added to deets. That should guarantee disjoint sets in such a way that we can't match on an item in both this function and the function to merge assist base details

Reply:
This is to support below case. 

**4. D1 is updated between C1 & F2. D1 does not move between F2 & C3**
    1. Since D1 didn’t move, it won’t be redownloaded as F2 has it and kopia will think it’s cached. But the input collections will have D1 because it changed between C1 (where we got the delta token from) and C3
    2. need to add logic to determine that D1 was indeed cached and pull the extension output for it from F2 during details merging
    3. incorrect to use D1 from C1 because it was changed in F2 → need to source from F2